controllers/feeds/reactionList.js

```diff
@@ -6,21 +6,17 @@ const {handleAnonymousData} = require('../../utils');
 const {User} = require('../../databases/models');
 const UsersFunction = require('../../databases/functions/users');
 
+const INITIAL_COMMENT_LEVEL = 1;
+const MAX_COMMENT_LEVEL = 3;
+
 module.exports = async (req, res) => {
   try {
     const {params, query} = req;
     const post = await Getstream.feed.getPlainFeedById(params.id);
     const myAnonymousId = await getAnonymUser(req.userId);
     const reaction = await reactionList(params.id, query.kind, query.limit);
-    let sortByDate = reaction.results.sort(
-      (a, b) => moment(a.created_at).unix() - moment(b.created_at).unix()
-    );
-    sortByDate = sortByDate.map((commentRes) => {
-      const sortComment = commentRes?.latest_children?.comment?.sort(
-        (a, b) => moment(a.created_at).unix() - moment(b.created_at).unix()
-      );
-      return {...commentRes, latest_children: {comment: sortComment}};
-    });
+
+    let level1Comments = buildComment(INITIAL_COMMENT_LEVEL, reaction.results);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The error handling seems to be missing in this section. If any of the async functions throw an error, it will cause the entire function to fail. It would be better to wrap each async call in a try-catch block to handle potential errors individually.
```diff
  const {params, query} = req;
+ try {
    const post = await Getstream.feed.getPlainFeedById(params.id);
+ } catch (error) {
+   return res.status(500).json({ error: 'Error fetching feed by id' });
+ }
+ try {
    const myAnonymousId = await getAnonymUser(req.userId);
+ } catch (error) {
+   return res.status(500).json({ error: 'Error fetching anonymous user' });
+ }
+ try {
    const reaction = await reactionList(params.id, query.kind, query.limit);
+ } catch (error) {
+   return res.status(500).json({ error: 'Error fetching reaction list' });
+ }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->